src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -275,7 +321,7 @@ internal void SetValidityBit(long index, bool value)
             Debug.Assert(bitMapBuffer.Length >= bitMapBufferIndex);
             if (bitMapBuffer.Length == bitMapBufferIndex)
                 bitMapBuffer.Append(0);
-            SetValidityBit(bitMapBuffer.RawSpan, (int)index, value);
+            SetValidityBit(bitMapBuffer.Span, (int)index, value);
```

Comment:
Is there a reason for this change?

Reply:
Yup. `Span` is the right(and more conservative) API to call. This was a bad fix(for a subtle bug). The `Span` property really returns `bitMapBuffer.Span.Slice(0, Length)`. Before this patch, when we encountered empty null bitmaps from Arrow, `Span` would throw. While debugging, I didn't realize that `Length = 0` was the bug. Instead I always looked at the `NullBitMapBuffer` and because they are always default initialized and the bitmaps encode 8 indices into 1 byte, I thought calling `Span.Slice(0, Length)` was the bug and called `RawSpan` instead. It worked because say `RawSpan[3]` really exists and doesn't throw, and because the `NullCount` was 0, the unit tests passed. That didn't address the `Length = 0` bug at all. With this patch, we always initialize the buffers and **set** the bits. So `Length` is now set to the length of the column and `Span` doesn't throw. The new test cases in `TestEmptyArrowColumns` explicitly test for this.